chore: Switch Google web sign-in to the redirect flow (#497)
serverpod/serverpod_docs: docs/06-concepts/11-authentication/04-providers/03-google/01-setup.md

      "web": {
        "client_id": "your-client-id.apps.googleusercontent.com",
        "client_secret": "your-client-secret",
        "redirect_uris": ["http://localhost:8082/auth/callback"]
      }
    }
```

> Replace `your-client-id` and `your-client-secret` with the values from the Google Auth Platform. The `redirect_uris` must match the **Authorized redirect URIs** you configured in the previous step. For web sign-in, this is the route URL from [Web setup](#web). If you run `flutter run -d chrome` locally, use the `auth.html` URL from the [separately-hosted Flutter web](./customizations#separately-hosted-flutter-web) fallback instead.

Reviewer: The user is asked to enter redirect uris here, while being adviced to skip it above and instead, if using web, enter the uris in the web section below. Make the user flow consistent.

Author: Fixed, oauth client step now asks for the URIs upfront

> Replace `your-client-id` and `your-client-secret` with the values from the Google Auth Platform. For iOS- or Android-only setups, you can leave `redirect_uris` as an empty array (`[]`); web sign-in requires the route URL from [Web setup](#web), which is the same value you add to **Authorized redirect URIs** on the OAuth client. If you run `flutter run -d chrome` locally, use the `auth.html` URL from the [separately-hosted Flutter web](./customizations#separately-hosted-flutter-web) fallback instead.

Reviewer 2: here `iOS-` isn't typo?

Author: Updated to 'iOS-only or Android-only'

### Web

> On web, Google signs the user in through an OAuth2 redirect: the browser is sent to `accounts.google.com`, the user signs in, and Google redirects back to a callback page that posts the result back to your Flutter app via `postMessage`. Serverpod hosts this callback for you when the Flutter web app is served by the same Serverpod server (the default project template, which copies the Flutter web build into Serverpod's `web/app/` directory via the `flutter_build` script).

Reviewer: This part was a bit confusing. In this guide the users don't need to know how the nuts and bolts work under the hood. Simplified the language.

```suggestion
On web, Google signs the user in through an OAuth2 redirect: the browser is sent to `accounts.google.com`, the user signs in, and Google redirects back to a callback page that posts the result back to your Flutter app via `postMessage`. 

Serverpod can host this callback as the Flutter web app is served by it.
```

Author: Applied the suggestion

### Web

On web, Google signs the user in through an OAuth2 redirect: the browser is sent to `accounts.google.com`, the user signs in, and Google redirects back to a callback page that posts the result back to your Flutter app via `postMessage`.

Serverpod can host this callback as the Flutter web app is served by it.

> 1. In `server.dart`, before `pod.start()`, register the callback route:

Reviewer 2: Question: what do you think if to replace `In server.dart` to `In server project`?

Author: Updated to 'In your server project's `server.dart` file' so the file is named with project context

   The route is provider-agnostic. Register it once and reuse the same callback URL across every OAuth2 PKCE provider (Google, GitHub, etc.).

2. Update the server OAuth client you created in the [previous section](#create-the-server-oauth-client-web-application) and add the following:

   - **Authorized JavaScript origins**: your Flutter web app's origin (e.g., `https://my-awesome-project.serverpod.space`).
   - **Authorized redirect URIs**: the full URL of the route you registered (e.g., `https://my-awesome-project.serverpod.space/auth/callback`).

   ![Web credentials configuration](/img/authentication/providers/google/2-credentials.png)

3. Pass the same URL to `initializeGoogleSignIn` via the `redirectUri` argument when you initialize the client (covered in [Initialize the Google sign-in service](#initialize-the-google-sign-in-service) below).

   :::tip
   To keep this URL out of `main.dart` and vary it per build, read it from `--dart-define`. See [Configuring the Web redirect URI](./customizations#configuring-the-web-redirect-uri) for the pattern.

>    :::

Reviewer 2: This tip is confusing me. 
Question: What if we remove `To keep this URL out of main.dart and vary it per build`, and optionally suggest saying `Also you can pass redirect URL from --dart-define`.

Author: Simplified to 'You can also pass the redirect URI via --dart-define'

:::note
If you develop locally with `flutter run -d chrome` (Flutter dev server runs on a separate port from Serverpod) or host the Flutter web build on a CDN separate from Serverpod, the route can't reach your Flutter app and you need the static `auth.html` flow instead. See [Separately-hosted Flutter web](./customizations#separately-hosted-flutter-web) for the fallback.
:::

## Present the authentication UI

### Initialize the Google sign-in service

> Open your Flutter app's `main.dart` (e.g., `my_project_flutter/lib/main.dart`). The Serverpod template already creates the `Client` and calls `client.auth.initialize()` inside `main()`. Add `client.auth.initializeGoogleSignIn()` on the line immediately after it:

Reviewer 2: Question: What if we replaced this by: 
```
In flutter project call this methods in `main` method. 

```dart
client.auth.initialize();
client.auth.initializeGoogleSignIn();
```
```

Author: Trimmed to just the lines that change, also dropped the explainer paragraph

   ```bash
   flutter run -d chrome --web-hostname localhost --web-port=49660
   ```

   - `-d chrome`: Run on the Chrome browser.
   - `--web-hostname localhost`: Bind to localhost.
   - `--web-port=49660`: Use a fixed port (pick any available port). This is the value you will add to **Authorized JavaScript origins** and **Authorized redirect URIs** in the next step.

3. **Update the server OAuth client.** Go back to the server OAuth client you created in the [previous section](#create-the-server-oauth-client-web-application) and add the following:

   - **Authorized JavaScript origins**: your Flutter web app's origin. For local development, `http://localhost:49660` (or whichever port you chose). For production, your Flutter web app's domain (e.g., `https://my-awesome-project.serverpod.space`).
   - **Authorized redirect URIs**: the full URL where `auth.html` is served. For local development, `http://localhost:49660/auth.html`. For production, the same URL with your production origin (e.g., `https://my-awesome-project.serverpod.space/auth.html`).

   ![Web credentials configuration](/img/authentication/providers/google/2-credentials.png)

   :::note

>    On the standard Serverpod project template, the Flutter web build is copied into the Serverpod web server's `web/app/` folder when you run the `flutter_build` script. If you deploy that way, the production callback lives at `https://your-domain.com/app/auth.html`, not at the domain root. Register that exact URL in **Authorized redirect URIs**.

Reviewer 2: Will be nice, if you mentioned `serverpod run flutter_build`.

Author: Will do great.
